### DISCUSSION TITLE
[QUESTION] Express Session

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [wobsoriano/trpc-nuxt#96] [Q&A] [answered]
Amazing package! It finally helped me understand tRPC. I just proposed in the official trpc repo to include this package in an "Integrations" section.

If anyone can give me a tip, how to integrate express-session in Nuxt 3?

In a pure Node project I would add an express app and install the session as a middleware.

But I don't seem to find a way to do it with the Nuxt 3 module.

Any tip will be appreciated. 

Wishing you all health, peace, and success! 🔥❤

**Comment by angelhdzmultimedia** [comment]
### Update

I just learned about nuxt-iron-session and h3-session.

I'm going to try h3-session first.

How to integrate it with trpc-nuxt?

**Reply by wobsoriano** [reply, marked as answer]
Check the [tRPC Context docs](https://trpc.io/docs/server/context) and if youre using `nuxt-iron-session` for example, you can return the session object in the context:

```ts
export default createNuxtApiHandler({
  router: appRouter,
  createContext: (event) => {
    return {
      session: event.context.session
    }
  },
})
```

Combine it with [trpc-shield](https://github.com/omar-dulaimi/trpc-shield) :)
